apps/src/codebridge/FileBrowser/hooks/useHandleFileUpload.ts

```diff
@@ -39,30 +38,22 @@ export const useHandleFileUpload = (
       // So don't attach other handlers to this button.
       document.body.click();
 
-      if (!isValidFileName(fileName)) {
-        dialogControl?.showDialog({
-          type: DialogType.GenericAlert,
-          title: codebridgeI18n.invalidNameError(),
-        });
-
-        sendCodebridgeAnalyticsEvent(
-          EVENTS.CODEBRIDGE_UPLOAD_INVALID_FILE_NAME,
-          appName,
-          {fileName}
-        );
-        return;
-      }
-      const duplicate = checkForDuplicateFilename({
+      const validationError = validateFileName({
         fileName,
         folderId,
         projectFiles,
         isStartMode,
         validationFile,
       });
-      if (duplicate) {
+
+      if (validationError) {
         dialogControl?.showDialog({
           type: DialogType.GenericAlert,
-          title: duplicate,
+          title: validationError,
+        });
+        sendCodebridgeAnalyticsEvent(EVENTS.CODEBRIDGE_UPLOAD_FAILED, appName, {
+          fileName,
+          error: validationError,
```

Comment:
now a failed upload will send the standard `UPLOAD_FAILED` event, with more information as to exactly what happened.